testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/jca/datasource/JdbcDriversInWarTestCase.java

```diff
@@ -0,0 +1,138 @@
+/*
+ * Copyright The WildFly Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.jboss.as.test.integration.jca.datasource;
+
+import org.jboss.arquillian.container.test.api.Deployment;
+import org.jboss.arquillian.container.test.api.RunAsClient;
+import org.jboss.arquillian.junit.Arquillian;
+import org.jboss.as.controller.client.helpers.Operations;
+import org.jboss.as.controller.descriptions.ModelDescriptionConstants;
+import org.jboss.as.test.integration.jca.JcaMgmtBase;
+import org.jboss.dmr.ModelNode;
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.spec.JavaArchive;
+import org.jboss.shrinkwrap.api.spec.WebArchive;
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import java.sql.Driver;
+
+import static org.jboss.as.controller.descriptions.ModelDescriptionConstants.ADD;
+import static org.jboss.as.controller.descriptions.ModelDescriptionConstants.OP;
+import static org.jboss.as.controller.descriptions.ModelDescriptionConstants.OP_ADDR;
+import static org.jboss.as.controller.descriptions.ModelDescriptionConstants.SUBSYSTEM;
+
+/**
+ * Test the situation when multiple JDBC drivers inside a war.
+ *
+ * @author <a href="mailto:lgao@redhat.com">Lin Gao</a>
+ */
+@RunWith(Arquillian.class)
+@RunAsClient
+public class JdbcDriversInWarTestCase extends JcaMgmtBase {
+
+    private static final String DEPLOYMENT = "jdbc-in-war.war";
+
+    @Deployment(name = DEPLOYMENT)
+    public static WebArchive jdbcInWarDeployment() throws Exception {
+        WebArchive war = ShrinkWrap.create(WebArchive.class, DEPLOYMENT);
+        war.addAsLibrary(ShrinkWrap.create(JavaArchive.class, "testDriver1.jar")
+                .addClasses(TestDriver.class, DummyDataSource.class, DummyXADataSource.class)
+                .addAsServiceProviderAndClasses(Driver.class, TestDriver.class));
+        war.addAsLibrary(ShrinkWrap.create(JavaArchive.class, "testDriver2.jar")
+                .addClasses(TestDriver2.class, DummyDataSource.class, DummyXADataSource.class)
+                .addAsServiceProviderAndClasses(Driver.class, TestDriver2.class));
+        return war;
+    }
+
+    @Test
+    public void testJdbcDrivers() throws Exception {
+        String driverName = DEPLOYMENT;
```

Comment:
@gaol maybe it will also be useful to verify the number of deployed drivers?

Reply:
right, good catch ! 
will try to get the jdbc driver using short name and it should fail.
checking number of deployed jdbc drivers maybe confusing as we need to filter out the default `h2` driver.